### PR TITLE
features: setup questions screen

### PR DESCRIPTION
**Pull request by PanVova**

### CODE REVIEW

File: app/src/main/java/com/panhuk/quizapp/MainActivity.kt [hunk 1]
```diff
@@ -6,6 +6,7 @@ import androidx.appcompat.app.AppCompatDelegate
 import androidx.lifecycle.lifecycleScope
 import androidx.navigation.NavController
 import androidx.navigation.findNavController
+import com.example.setupquestionfeature.SetupQuestionsNavigator
```
Comment (NickHuk) [inline app/src/main/java/com/panhuk/quizapp/MainActivity.kt]: package name xD

File: app/src/main/java/com/panhuk/quizapp/MainActivity.kt [hunk 2]
```diff
@@ -56,8 +57,8 @@ class MainActivity() : AppCompatActivity(), MenuNavigator, FirstTimeNavigator, F
     navigator.navigate(R.id.firstTimeFragment)
   }
 
-  override fun navigateMenuToPlayFragment() {
-    navigator.navigate(R.id.menu_play)
+  override fun navigateMenuToSetupQuestionsFragment() {
+    navigator.navigate(R.id.menu_setup_questions)
```
Comment (NickHuk) [inline app/src/main/java/com/panhuk/quizapp/MainActivity.kt]: navController is a better name for this object

File: features/playFeature/src/main/java/com/panhuk/playfeature/di/PlayFactory.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      category: Int?,
+      difficulty: String?,
+      questionNumber: String?,
+      type: String?
+    ): ViewModelProvider.Factory = object : ViewModelProvider.Factory {
+      override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return assistedFactory.create(category, difficulty, questionNumber, type) as T
+      }
+    }
+  }
```
Comment (NickHuk) [inline features/playFeature/src/main/java/com/panhuk/playfeature/di/PlayFactory.kt:34]: hmm,  interesting ...

File: features/setupQuestionsFeature/src/main/java/com/example/setupquestionfeature/SetupQuestionsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      }
+    }
+  }
+
+  fun updateQuestion(title: Int, text: String) {
+    when (title) {
+      R.string.number_of_questions -> question = text
+      R.string.difficulty -> difficulty = text
+      R.string.type -> type = text
+      R.string.category -> {
```
Comment (NickHuk) [inline features/setupQuestionsFeature/src/main/java/com/example/setupquestionfeature/SetupQuestionsViewModel.kt]: that's a bad approach, use enum or sealed class to define this behavior but not android resource ids

**Review by NickHuk — APPROVED** [review]

File: app/src/main/java/com/panhuk/quizapp/MainActivity.kt (hunk 1 above)
Comment (PanVova) [reply app/src/main/java/com/panhuk/quizapp/MainActivity.kt]: +

**Pull request merged**